trie/zk/tree_path.go

```diff
@@ -65,3 +84,40 @@ func (p TreePath) toHexBytes() []byte {
 	}
 	return bytes
 }
+
+func (p TreePath) NextPath() TreePath {
+	for i := len(p) - 1; i >= 0; i-- {
+		if p[i] == 0x0 {
+			p[i] = 0x1
+			break
+		} else if p[i] == 0x1 {
+			p[i] = 0x0
+		} else {
+			log.Panicf("invalid tree path %v\n", p)
```

Comment:
If it's an invalid tree path, shouldn't throw it?

Reply:
Invalid tree paths can only be the result of bad development, so I panicked.
By the way, since I introduced the `iteratorkey` concept, that function is only used in test code.